src/Mod/Arch/importWebGL.py

```diff
@@ -18,233 +19,782 @@
 #*   USA                                                                   *
 #*                                                                         *
 #***************************************************************************
+#
+# REFS:
+# https://github.com/mrdoob/three.js/blob/master/examples/webgl_interactive_buffergeometry.html
+# https://threejs.org/examples/#webgl_buffergeometry_lines
+# https://forum.freecadweb.org/viewtopic.php?t=51245
+# https://forum.freecadweb.org/viewtopic.php?t=29487
+# https://threejs.org/examples/#webgl_raycast_sprite
+#
+# Params for export()
+#    'colors' is of the form: {'Body': [1,0,0], 'Body001': [1,1,0], 'Body002': [1,0,1] }
+#    'camera' is of the form: "PerspectiveCamera {\n  viewportMapping ADJUST_CAMERA\n  position 30.242626 -51.772324 85.63475\n  orientation -0.4146691 0.088459305 -0.90566254  4.7065201\nnearDistance 53.126431\n  farDistance 123.09125\n  aspectRatio 1\n  focalDistance 104.53851\n  heightAngle 0.78539819\n\n}"
+#    The 'camera' string for the active document may be generated from: import OfflineRenderingUtils; OfflineRenderingUtils.getCamera(FreeCAD.ActiveDocument.FileName);
+#
+# Development reload oneliner:
+# def re(): from importlib import reload;import importWebGL;reload(importWebGL);o=FreeCAD.getDocument("curve");importWebGL.export([o.getObject("Body")],u"C:/Users/Travis/Desktop/test.htm");
 
-"""FreeCAD webgl exporter
-
-options: importWebGL.wireframeStyle = "faceloop" (can also be "multimaterial" or None)
-importWebGL.template = a complete html file, where $CameraData is a placeholder for the 
-FreeCAD camera, and $ObjectsData a placeholder for the FreeCAD objects.
-importWebGL.linewidth = an integer, specifying the width of lines in "faceloop" mode"""
+"""FreeCAD WebGL Exporter"""
 
-import FreeCAD,Draft,Part,DraftGeomUtils
+import FreeCAD,Mesh,Draft,Part,DraftGeomUtils,Arch,OfflineRenderingUtils,json,six
 
 if FreeCAD.GuiUp:
     import FreeCADGui
     from DraftTools import translate
 else:
     FreeCADGui = None
-    # \cond
-    def translate(ctxt,txt,utf8_decode=True):
-        return txt
-    # \endcond
-
-## @package importWebGL
-#  \ingroup ARCH
-#  \brief WebGL file format exporter
-#
-#  This module provides tools to export HTML files containing the 
-#  exported objects in WebGL format and a simple three.js-based viewer.
```

Comment:
Doxygen comments were removed but never replaced

Reply:
This has been added back in